apis/python/tests/test_indexed_dataframe.py

```diff
@@ -289,6 +294,267 @@ def make_dataframe(request):
 )
 def test_index_types(tmp_path, make_dataframe):
     """Verify that the index columns can be of various types"""
-    sdf = soma.IndexedDataFrame(tmp_path.as_posix())
-    sdf.create(make_dataframe.schema, index_column_names=["index"])
-    sdf.write(make_dataframe)
+    sidf = soma.IndexedDataFrame(tmp_path.as_posix())
+    sidf.create(make_dataframe.schema, index_column_names=["index"])
+    sidf.write(make_dataframe)
+
+
+def make_multiply_indexed_dataframe(tmp_path, index_column_names: List[str]):
+    """
+    Creates a variably-indexed IndexedDataFrame for use in tests below.
+    """
+    schema = pa.schema(
+        [
+            # TO DO: Support non-int index types when we have non-int index support
+            # in libtiledbsoma's SOMAReader. See also
+            # https://github.com/single-cell-data/TileDB-SOMA/issues/418
+            # https://github.com/single-cell-data/TileDB-SOMA/issues/419
+            ("index1", pa.int64()),
+            ("index2", pa.int64()),
+            ("index3", pa.int64()),
+            ("index4", pa.int64()),
+            ("soma_joinid", pa.int64()),
+            ("A", pa.int64()),
+        ]
+    )
+
+    sidf = soma.IndexedDataFrame(uri=tmp_path.as_posix())
+    sidf.create(schema=schema, index_column_names=index_column_names)
+
+    data = {
+        "index1": [0, 1, 2, 3, 4, 5],
+        "index2": [400, 400, 500, 500, 600, 600],
+        "index3": [0, 1, 0, 1, 0, 1],
+        "index4": [1000, 2000, 1000, 1000, 1000, 1000],
+        "soma_joinid": [10, 11, 12, 13, 14, 15],
+        "A": [10, 11, 12, 13, 14, 15],
+    }
+
+    n_data = len(data["index1"])
+    rb = pa.Table.from_pydict(data)
+    sidf.write(rb)
+    return (schema, sidf, n_data)
+
+
+@pytest.mark.parametrize(
+    "io",
+    [
```

Comment:
missing several important tests:
* ranges with a step != 1,  e.g.,   ids=[range(0,10,2)]
* empty ids list or tuple, e.g.,   ids=[]
* slices with step != 1
* slices with negative indexing
there may be more - this was a quick look....

Reply:
Oh, I would add some additional expected errors, e.g.,  array of floats or other unsupported primitives, datetime, etc.